igzip_threaded cannot append

An `xopen` test fails when isal 1.5.2 is installed. The problem can be reproduced thus:
```python
from isal.igzip_threaded import open as isal_open

with isal_open("out.gz", "wb") as f:
    f.write(b"AB")
with isal_open("out.gz", "ab") as f:
    f.write(b"CD")
with isal_open("out.gz", "rb") as f:
    contents = f.read()

assert contents == b"ABCD", contents
```
Fails with:
```
Traceback (most recent call last):
  File ".../test.py", line 10, in <module>
    assert contents == b"ABCD", contents
AssertionError: b'CD'
```
The assertion is not triggered with version 1.5.1.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -7,6 +7,11 @@ Changelog
 .. This document is user facing. Please word the changes in such a way
 .. that users understand how the changes affect the new version.
 
+version 1.5.3
+-----------------
++ Fix a bug where append mode would not work when using
+  ``igzip_threaded.open``.
+
 version 1.5.2
 -----------------
 + Fix a bug where a filehandle remained opened when ``igzip_threaded.open``

--- a/src/isal/igzip_threaded.py
+++ b/src/isal/igzip_threaded.py
@@ -63,6 +63,7 @@ def open(filename, mode="rb", compresslevel=igzip._COMPRESS_LEVEL_TRADEOFF,
         gzip_file = io.BufferedWriter(
             _ThreadedGzipWriter(
                 filename,
+                mode.replace("t", "b"),
                 block_size=block_size,
                 level=compresslevel,
                 threads=threads
@@ -197,11 +198,16 @@ class _ThreadedGzipWriter(io.RawIOBase):
     """
     def __init__(self,
                  filename,
+                 mode: str = "wb",
                  level: int = isal_zlib.ISAL_DEFAULT_COMPRESSION,
                  threads: int = 1,
                  queue_size: int = 1,
                  block_size: int = 1024 * 1024,
                  ):
+        if "t" in mode or "r" in mode:
+            raise ValueError("Only binary writing is supported")
+        if "b" not in mode:
+            mode += "b"
         self.lock = threading.Lock()
         self.exception: Optional[Exception] = None
         self.level = level
@@ -238,7 +244,7 @@ class _ThreadedGzipWriter(io.RawIOBase):
         self.running = False
         self._size = 0
         self._closed = False
-        self.raw = open_as_binary_stream(filename, "wb")
+        self.raw = open_as_binary_stream(filename, mode)
         self._write_gzip_header()
         self.start()
 

--- a/tests/test_igzip_threaded.py
+++ b/tests/test_igzip_threaded.py
@@ -169,3 +169,25 @@ def test_writer_write_after_close(threads):
     with pytest.raises(ValueError) as error:
         f.write(b"abc")
     error.match("closed")
+
+
+def test_igzip_threaded_append(tmp_path):
+    test_file = tmp_path / "test.txt.gz"
+    with igzip_threaded.open(test_file, "wb") as f:
+        f.write(b"AB")
+    with igzip_threaded.open(test_file, mode="ab") as f:
+        f.write(b"CD")
+    with gzip.open(test_file, "rb") as f:
+        contents = f.read()
+    assert contents == b"ABCD"
+
+
+def test_igzip_threaded_append_text_mode(tmp_path):
+    test_file = tmp_path / "test.txt.gz"
+    with igzip_threaded.open(test_file, "wt") as f:
+        f.write("AB")
+    with igzip_threaded.open(test_file, mode="at") as f:
+        f.write("CD")
+    with gzip.open(test_file, "rt") as f:
+        contents = f.read()
+    assert contents == "ABCD"

